### PR TITLE
Feature/use fetch

### PR DESCRIPTION
**Pull request by F-Kirchhoff**
- add useFetch hook
- implement board data fetch
- implement topics data fetch

### CODE REVIEW

File: src/app/App.tsx [hunk 1]
```diff
@@ -1,85 +1,38 @@
-import React, { useState } from 'react'
-import { BrowserRouter, Routes, Route } from 'react-router-dom'
+
+import React, { useEffect, useState } from 'react'
+import { BrowserRouter, Routes, Route, useParams } from 'react-router-dom'
+import TopicView from './pages/TopicView/TopicView'
 import Dashboard from './pages/Dashboard/Dashboard'
 
 import type { Need, Topic } from './types/types'
+import type { ObjectId } from 'mongodb'
```
Comment (phlgr) [inline src/app/App.tsx:8]: Don't use this type in the frontend. I would suggest just declaring the id as a string

File: src/app/App.tsx [hunk 2]
```diff
@@ -1,85 +1,38 @@
...
+  _id: ObjectId
+  name: string
+  topics: Topic[]
+}
 
 function App(): JSX.Element {
-  const [topics, setTopics] = useState(TOPICS)
+  const [board, fetchBoard] = useFetch<Board>('api/boards/exampleboard')
+  const [DBtopics, fetchTopics] = useFetch<Topic[]>('api/topics/')
+
```
Comment (phlgr) [inline src/app/App.tsx:22]:
```suggestion
  const [board, fetchBoard] = useFetch<Board>('/api/boards/exampleboard')
  const [DBtopics, fetchTopics] = useFetch<Topic[]>('/api/topics/')

```

File: src/server/routes/topics.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { ObjectId } from 'mongodb'
 const topics = express.Router()
 
 topics
-  .get('/many', async (req: Request, res: Response) => {
+  .post('/', async (req: Request, res: Response) => {
```
Comment (phlgr) [inline src/server/routes/topics.ts:9]: Please don't chain these methods.

File: tests.http [hunk 4]
```diff
@@ -0,0 +1,9 @@
+### GET many topcis
+GET http://localhost:3001/api/topics/many
```
Comment (phlgr) [inline tests.http:2]: This route does not exist does it?

File: src/app/hooks/useFetch.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+      case 'POST': {
+        const res = await fetch(url, {
+          method,
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: body ? JSON.stringify(body) : '[]',
+        })
+        if (res.status === 200) {
```
Comment (phlgr) [inline src/app/hooks/useFetch.tsx:29]:
```suggestion
        if (res.ok) {
```

File: src/app/hooks/useFetch.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+function useFetch<Type>(
+  url: string
+): [Type | null, (method: string, body?: any) => void] {
+  const [data, setData] = useState(null)
+
+  async function fetchData(method: string, body?: any) {
+    switch (method) {
+      case 'GET': {
+        const res = await fetch(url)
+        if (res.status === 200) {
```
Comment (phlgr) [inline src/app/hooks/useFetch.tsx:12]:
```suggestion
        if (res.ok) {
```

**Review by phlgr — CHANGES_REQUESTED** [review]

File: src/server/routes/topics.ts (hunk 3 above)
Comment (F-Kirchhoff) [reply src/server/routes/topics.ts:9]: the first one is removed...

**Pull request merged**